.github/workflows/ci.yml

```diff
@@ -57,40 +60,83 @@ jobs:
           ~/.cargo/registry/cache/
           ~/.cargo/git/db/
           target/
-        key: ${{ runner.os }}-cargo-${{ steps.args.outputs.args }}-${{ hashFiles('**/Cargo.lock') }}
+        key: ${{ runner.os }}-${{ runner.arch }}-cargo-${{ steps.args.outputs.args }}-${{ hashFiles('**/Cargo.lock') }}
     - run: cargo ${{ matrix.args }}
 
   container:
-    runs-on: ubuntu-24.04
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - amd64
+          - arm64
+    runs-on: ${{ (matrix.arch == 'arm64' && 'ubuntu-24.04-arm') || 'ubuntu-24.04' }}
     steps:
     - uses: actions/checkout@v4
       with:
         submodules: true
+
+    - run: |
+        echo "FACT_TAG=$(make tag)-${{ matrix.arch }}" >> "$GITHUB_ENV"
```

Comment:
Would it make sense to make `container` depend on `vars` and do something like this instead?
```yaml
  container:
    needs: vars
    env:
      FACT_TAG: ${{ needs.vars.outputs.tag }}-${{ matrix.arch }}
```